src/controllers/addTodo.js

```diff
@@ -0,0 +1,28 @@
+(function() {
```

Comment:
components can be used here:
.component('editTodo', {
+        templateUrl: 'app/todo-form/todo-form.template.html',
+        bindings: {
+            todo: '<'
+        },
+        controller: ['TodoService', '$location', function (TodoService, $location) {
+            var ctrl = this;
+
+            ctrl.upsertTodo = upsertTodo;
+            ctrl.back = back;

Reply:
components - will be used for second Task